JumpscaleCore/core/BASECLASSES/JSBase.py

```diff
@@ -627,25 +627,21 @@ def _parent_name_get(self):
     def _mother_id_get(self):
         """
         this goes to all parents till it finds a parent which has a model attached
-        this is to find the parent which acts as the mother for the children
+        this is to find the parent which acts as the mother for the children.
         when you do a search only the children of this mother will be shown
-        :return:
+        If mother was created with save=False, it won't have id but will still exist as a parent.
+
+        :rtype: tuple
+        :return: The id of the mother and a flag indicating if the child has a mother
         """
         obj = self
         while obj and obj._parent:
             if isinstance(obj._parent, j.baseclasses.object_config):
-                if obj._parent._id is None:
-                    if obj._parent.name is None:
-                        raise j.exceptions.JSBUG("cannot happen, there needs to be a name")
-                    else:
-                        obj._parent.save()
```

Comment:
We can't just save any object we find. It might have been created with `save=True`. 
I refactored the function to return a tuple of the parent_id and a boolean indicating that the object indeed has a parent. That way if the parent wasn't saved, we can still know that there is a parent.